File: src/test/java/features/stepdefs/CommonStepDef.java
package features.stepdefs;

import features.commons.Authentication;
import features.context.Context;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class CommonStepDef {

    @Given("When login with {string} {string}")
    public void login(String username, String password) throws IOException {
        Authentication.login(username, password);
    }

    @Then("Client will receives status code of {int}")
    public void expects_status_code_of(int statusCode) {
        assertEquals(statusCode, Context.response.code());
    }
}

File: src/test/kotlin/features/stepdefintions/CommonStepDef.kt
package features.stepdefintions

import features.common.Auth
import features.context.TestContext
import io.cucumber.java.en.Given
import io.cucumber.java.en.Then

class CommonStepDef {

    @Given("When login with {string} {string}")
    fun login(username: String, password: String) {
        Auth.login(username, password)
    }

    @Then("Client will receives status code of {int}")
    fun expects_status_code_of(statusCode: Int) {
        val response = TestContext.response
        assert(response != null)
        assert(statusCode == response!!.code())
    }
}